### ISSUE TITLE
[eslint-plugin-lit-a11y]: optional chaining breaking rules

### ISSUE THREAD

**Issue by bennypowers** [open-wc/open-wc#1901]
```js
html`
  <a href="${launch?.links?.article_link}" target="_blank" rel="noopener noreferer">Media</a>
  <a href="${launch?.links?.wikipedia}" target="_blank" rel="noopener noreferer">Wikipedia</a>
`;
```
> The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md" eslint(lit-a11y/anchor-is-valid)

Whereas these pass:
```js
html`
  <a href="${article_link}" target="_blank" rel="noopener noreferer">Media</a>
  <a href="${wikipedia}" target="_blank" rel="noopener noreferer">Wikipedia</a>
`;
```

**Comment by thepassle** [comment]
Think this might be a missing case here:

https://github.com/open-wc/open-wc/blob/cf1c52709a58c1b560c69cef01f209b88a166f15/packages/eslint-plugin-lit-a11y/template-analyzer/util.js#L235

But frankly I'm strongly considering reverting the logic we have there, I'm not sure if its worth it

**Comment by thepassle** [comment]
this was fixed in https://github.com/open-wc/open-wc/pull/1907
